templates/default/deploy-scripts/src/deployer.rs

```diff
@@ -0,0 +1,394 @@
+        ));
+
+        let tx = init_contract(
+            &*self.key,
+            self.key.address,
+            nonce.nonce,
+            expiry,
+            payload,
+            energy,
+        );
+
+        let bi = transactions::BlockItem::AccountTransaction(tx);
+
+        let tx_hash = self.client.clone().send_block_item(&bi).await?;
+
+        println!("Sent tx: {tx_hash}");
+
+        let (_, block_item) = self.client.wait_until_finalized(&tx_hash).await?;
+
+        let contract_address = self.check_outcome_of_initialization_transaction(&block_item)?;
+
+        println!(
+            "Transaction finalized: tx_hash={} contract=({}, {})",
+            tx_hash, contract_address.index, contract_address.subindex,
+        );
+
+        Ok((tx_hash, block_item, contract_address))
+    }
+
+    /// A function to update a smart contract instance on the chain.
+    ///
+    /// If successful, the transaction
+    /// hash, and the block item are returned.
+    ///
+    /// An optional energy for the transaction can be
+    /// given. If `None` is provided, 50000 energy is used as a default energy
+    /// value. An optional expiry time for the transaction can be given. If
+    /// `None` is provided, the local time + 300 seconds is used as a default
+    /// expiry time.
+    pub async fn update_contract(
+        &mut self,
+        update_payload: UpdateContractPayload,
+        energy: Option<GivenEnergy>,
+        expiry: Option<TransactionTime>,
+    ) -> Result<(TransactionHash, BlockItemSummary), Error> {
+        println!("\nUpdating contract....");
+
+        let nonce = self.get_nonce(self.key.address).await?;
+
+        if !nonce.all_final {
+            bail!("Nonce not final")
+        }
+
+        let payload = transactions::Payload::Update {
+            payload: update_payload,
+        };
+
+        let expiry = expiry.unwrap_or(TransactionTime::from_seconds(
+            (chrono::Utc::now().timestamp() + 300) as u64,
+        ));
+
+        let energy = energy.unwrap_or(GivenEnergy::Absolute(Energy { energy: 50000 }));
+
+        let tx = transactions::send::make_and_sign_transaction(
+            &*self.key,
+            self.key.address,
+            nonce.nonce,
+            expiry,
+            energy,
+            payload,
+        );
+        let bi = transactions::BlockItem::AccountTransaction(tx);
+
+        let tx_hash = self.client.clone().send_block_item(&bi).await?;
+
+        println!("Sent tx: {tx_hash}");
+
+        let (_, block_item) = self.client.wait_until_finalized(&tx_hash).await?;
+
+        self.check_outcome_of_update_transaction(&block_item)?;
+
+        println!("Transaction finalized: tx_hash={}", tx_hash,);
+
+        Ok((tx_hash, block_item))
+    }
+
+    /// A function to estimate the energy needed to send a transaction on the
+    /// chain.
+    ///
+    /// If successful, the transaction energy is returned by this function.
+    /// This function can be used to dry-run a transaction.
```

Comment:
Note that it's important to clarify here that this will only estimate **execution energy**.

This is the energy that should be supplied to the `update` function, but it's important to clarify that the actual NRG cost on the chain will also have the administrative overhead based on the size and number of transactions.